Assets/Scripts/Units/Javelin Unit/JavelinUnit.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace RogueApeStudio.Crusader.Units.JavelinUnit
+{
+    public class JavelinUnit : MonoBehaviour
+    {
+        [Header("Movement settings")]
+        [SerializeField] private float _stopDistance = 1.8f;
+        [SerializeField] private float _movementSpeed = 3.5f;
+        [SerializeField] private float _startAttackDistance = 50.0f;
+        [Header("Attack settings")]
+        [SerializeField] private float _attackRange = 500.0f;
+        [SerializeField] private Vector2 _projectileForce = new(32, 2);
+        [Header("Line Of Sight Seetings")]
+        [SerializeField] private int _amountOfRayCasts = 3;
+        [SerializeField] private float _unitWidth = 1f;
+        [SerializeField] private float _secondsLosNeeded = 1f;
+        [Header("Dependencies")]
+        [SerializeField] private UnitMovement _localUnitMovement;
+        [SerializeField] private Animator _localAnimator;
+        [SerializeField] private GameObject _projectile;
+        [SerializeField] private Transform _projectileSpawn;
+
+        private IJavelinUnitState _currentState;
+        private IJavelinUnitState[] _states;
+
+        public UnitMovement LocalUnitMovement => _localUnitMovement;
+        public Animator LocalAnimator => _localAnimator;
+        public float StartAttackDistance => _startAttackDistance;
+        public float AttackRange => _attackRange;
+        public GameObject Projectile => _projectile;
+        public Transform ProjectileSpawn => _projectileSpawn;
+        public Vector2 ProjectileForce => _projectileForce;
+        public float SecondsLosNeeded => _secondsLosNeeded;
+
+
+        private void Awake()
+        {
+            int statesAmount = Enum.GetNames(typeof(JavelinUnitStateId)).Length;
+            _states = new IJavelinUnitState[statesAmount];
+            AddJavelinUnitState(new JavelinUnitChaseState());
+            AddJavelinUnitState(new JavelinUnitAttackState());
+            _currentState = GetJavelinUnitState(JavelinUnitStateId.Chase);
+            LocalUnitMovement.SetStopDistance(_stopDistance);
+            LocalUnitMovement.SetSpeed(_movementSpeed);
+        }
+
+        void Update()
+        {
+            _currentState.UpdateState(this);
```

Comment:
Does this need to run _each_ frame?

Reply:
yes

Reply:
I'll make an issue on it, but that's for post-dragons den.